swarm_transport.go

```diff
@@ -0,0 +1,61 @@
+package swarm
+
+import (
+	"fmt"
+	"strings"
+
+	transport "github.com/libp2p/go-libp2p-transport"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// TransportForAddr retrieves the appropriate transport for the given multiaddr.
```

Comment:
Would be nice to note that this implements the `Network` interface from `go-libp2p-transport` as it has nice docs

Reply:
Fair enough.